src/Api/Fields/ReportApiFields.php

```diff
@@ -0,0 +1,109 @@
+     * @param string|UriInterface $uri
+     */
+    public function completionWebhook(string $uri): void
+    {
+        $this->keyValuePairs["on_completion"] = $uri;
+    }
+
+    /** Business name, some checks will not work without this, e.g Local presence, Reviews. */
+    public function name(string $name): void
+    {
+        $this->keyValuePairs["name"] = $name;
+    }
+
+    /** Business phone number, some checks will not work without this, e.g Local presence, Reviews. */
+    public function phone(string $phone): void
+    {
+        $this->keyValuePairs["phone"] = $phone;
+    }
+
+    /**
+     * @param string $firstLine First line of business address, some checks will not work without this, e.g Local presence, Reviews
+     * @param string $buildingNameOrNumber Building number, enhances accuracy in some checks, e.g Local presence, Reviews
+     * @param string $street Street, enhances accuracy in some checks, e.g Local presence, Reviews
+     * @param string $city City, enhances accuracy in some checks, e.g Local presence, Reviews
+     * @param string $stateOrCounty State or county, enhances accuracy in some checks, e.g Local presence, Reviews
+     * @param string $zipOrPostcode Zip or postcode, enhances accuracy in some checks, e.g Local presence, Reviews
+     * @param string $countryCode ISO 2 letter code – Country, enhances accuracy in some checks, e.g Local presence, Reviews
+     */
+    public function address(
+        string $firstLine = "",
+        string $buildingNameOrNumber = "",
+        string $street = "",
+        string $city = "",
+        string $stateOrCounty = "",
+        string $zipOrPostcode = "",
+        string $countryCode = ""
+    ): void {
+        if($firstLine) {
+            $this->keyValuePairs["address"] = $firstLine;
+        }
+        if($buildingNameOrNumber) {
+            $this->keyValuePairs["number"] = $buildingNameOrNumber;
+        }
+        if($street) {
+            $this->keyValuePairs["street"] = $street;
+        }
+        if($city) {
+            $this->keyValuePairs["city"] = $city;
+        }
+        if ($stateOrCounty) {
+            $this->keyValuePairs["state"] = $stateOrCounty;
+        }
+        if($zipOrPostcode) {
+            $this->keyValuePairs["zip"] = $zipOrPostcode;
+        }
+        if($countryCode) {
+            $this->keyValuePairs["country_code"] = $countryCode;
+        }
+    }
+
+    /** Latitude and longitude, enhances accuracy in some checks, e.g Local presence, Reviews. */
+    public function latLng(float $lat, float $lon): void {
+        $this->keyValuePairs["lat"] = $lat;
+        $this->keyValuePairs["lng"] = $lon;
+    }
+
+    /**
+     * Products and services this business offers, some checks will not work without this, e.g Content keywords.
+     * @param string...$products Individual products and services passed as variable arguments
+     */
+    public function products(string...$products): void {
```

Comment:
I think there's meant to be a space between this `string ...$products`.

This one is a matter of style so just a talking point, but I tend to find variadic functions a little less clear than the array based alternatives. It also removes your ability to make any non-BC changes to the function as now you're committed to not being able to add any other fields to the signature.

It always ends up feeling a bit "but why?" to me :P